time/src/main/java/net/jqwik/time/internal/properties/arbitraries/valueRanges/OfPrecision.java

```diff
@@ -34,4 +35,57 @@ public boolean isSet() {
 		return precisionSet;
 	}
 
+	public boolean valueWithPrecisionIsAllowed(int minutes, int seconds, int nanos) {
+		boolean retVal = true;
+		switch (precision) {
+			case HOURS:
+				retVal = minutes == 0;
+			case MINUTES:
+				retVal = retVal && seconds == 0;
+			case SECONDS:
+				retVal = retVal && nanos == 0;
+				break;
+			case MILLIS:
+				retVal = (nanos % 1_000_000) == 0;
+				break;
+			case MICROS:
+				retVal = (nanos % 1_000) == 0;
+		}
+		return retVal;
+	}
+
+	public Duration calculateMaxPossibleDuration() {
+		switch (precision) {
+			case HOURS:
+				return Duration.ofSeconds((Long.MAX_VALUE / (60 * 60)) * (60 * 60), 0);
+			case MINUTES:
+				return Duration.ofSeconds((Long.MAX_VALUE / 60) * 60, 0);
+			case MILLIS:
+				return Duration.ofSeconds(Long.MAX_VALUE, 999_000_000);
+			case MICROS:
+				return Duration.ofSeconds(Long.MAX_VALUE, 999_999_000);
+			case NANOS:
+				return Duration.ofSeconds(Long.MAX_VALUE, 999_999_999);
+			default:
+				return Duration.ofSeconds(Long.MAX_VALUE, 0);
+		}
+	}
+
+	public LocalTime calculateMaxPossibleLocalTime() {
+		switch (precision) {
+			case HOURS:
+				return LocalTime.of(23, 0, 0, 0);
+			case MINUTES:
+				return LocalTime.of(23, 59, 0, 0);
+			case MILLIS:
+				return LocalTime.of(23, 59, 59, 999_000_000);
+			case MICROS:
+				return LocalTime.of(23, 59, 59, 999_999_000);
+			case NANOS:
+				return LocalTime.of(23, 59, 59, 999_999_999);
+			default:
+				return LocalTime.of(23, 59, 59, 0);
+		}
+	}
```

Comment:
I don't know if it is possible to make it simpler without switch, but I think not.

Is the class now like you want or what should I do next?

Reply:
Looks good to me.

Reply:
Should I change it in all classes?

Example: `calculateEffectiveMin()`

Reply:
Unless you see a reason why calculate* is the better name in some cases